src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -0,0 +1,248 @@
+package chisel3.util
+
+import chisel3._
+
+import chisel3.internal.Builder
+import chisel3.experimental.SourceInfo
+import chisel3.internal.sourceinfo.{MemTransform, SourceInfoTransform}
+import scala.language.reflectiveCalls
+import scala.language.experimental.macros
+
+/** A bundle of signals representing a memory read port.
+  *
+  * @tparam tpe The data type of the memory port
+  * @param addrWidth The width of the address signal
+  */
+class MemoryReadPort[T <: Data](tpe: T, addrWidth: Int) extends Bundle {
+  val address = Input(UInt(addrWidth.W))
+  val enable = Input(Bool())
+  val data = Output(tpe)
+  val clock = Input(Clock())
+}
+
```

Comment:
I do not think that clocks should be part of the memory port Bundles. Yes, we may want to associate different clocks with different memory ports for the actual memory, but I don't think that putting them in these Bundles is the right API. If a user wants to bulk connect the memory port to a wire or something, they almost certainly are NOT going to want to be dealing with the clock, so I think it's just going to be an API burden in that case.

The API I have in mind (discussed with @aswaterman), is at the point of declaring the SRAM, you can describe the clocks, ie. whereas the normal API is:
```scala
  def apply[T <: Data](
     size:              BigInt,
     tpe:               T,
     numReadPorts:      Int,
     numWritePorts:     Int,
     numReadwritePorts: Int
   )
```
We can add one where instead of a number of ports, you give the clocks you want for each port:
```scala
  def apply[T <: Data](
     size:              BigInt,
     tpe:               T,
     readPortClocks:      Seq[Clock],
     writePortClocks:     Seq[Clock],
     readwritePortClocks: Seq[Clock]
   )
```
We can just have this PR be single clock and add that API in a follow on PR though.

Reply:
Sounds great.